File: app/src/main/java/com/example/cbanda/achieve/presentation/detail/GoalDetailViewModel.java
package com.example.cbanda.achieve.presentation.detail;

import android.app.Application;
import android.arch.lifecycle.AndroidViewModel;
import android.arch.lifecycle.LiveData;
import android.support.annotation.NonNull;

import com.example.cbanda.achieve.models.db.Goal;
import com.example.cbanda.achieve.models.repository.GoalRepository;
import com.example.cbanda.achieve.utilities.InjectorUtils;


public class GoalDetailViewModel extends AndroidViewModel {
    GoalRepository goalRepository;

    public GoalDetailViewModel(@NonNull Application application) {
        super(application);
        goalRepository = InjectorUtils.getInstance(application);
    }

    LiveData<Goal> getGoalById(int id) {
        return goalRepository.getGoalById(id);
    }
}

File: app/src/main/java/com/example/cbanda/achieve/presentation/detail/GoalDetailViewModel.kt
package com.example.cbanda.achieve.presentation.detail

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData

import com.example.cbanda.achieve.models.db.Goal
import com.example.cbanda.achieve.models.repository.GoalRepository
import com.example.cbanda.achieve.utilities.InjectorUtils


class GoalDetailViewModel(application: Application) : AndroidViewModel(application) {
    private val goalRepository: GoalRepository = InjectorUtils.getInstance(application)

    fun getGoalById(id: Int): LiveData<Goal> {
        return goalRepository.getGoalById(id)
    }
}
